### ISSUE TITLE
questions about this toolchain

### ISSUE THREAD

**Issue by houzi-** [lancethepants/tomatoware#53]
Hi guys

The binaries compiled by this toolchain should be based on ARMV7, right?

**Comment by lancethepants** [comment]
Ye

> Hi guys
> 
> The binaries compiled by this toolchain should be based on ARMV7, right?

Yes, armv7 Cortex-A9 softfloat.

**Comment by houzi-** [comment]
OK，Will it support armv8 Cortex-A9 in the future?

**Comment by lancethepants** [comment]
I would love to make an armv8/arm64/aarch64 toolchain version but I don't have any arm64 hardware.

**Comment by lancethepants** [comment]
What arm64 hardware are you running? Just out of curiosity.

**Comment by houzi-** [comment]
ASUS AX86U😬

**Comment by houzi-** [comment]
merlin

**Comment by lancethepants** [comment]
Apparently AsusWRT Merlin userspace tools are using 32bit with no vfpv3 or neon. There is a pending pull request to change this.
https://github.com/RMerl/asuswrt-merlin.ng/pull/804

**Comment by houzi-** [comment]
Yes, but the router system of the Merlin Broadcom solution is based on 64 kernels and is compatible with 32-bit and 64-bit binary operation.

**Comment by lancethepants** [comment]
Just bought me an rt-ax88u. So may be a few weeks, but I plan to get tomatoware working on this. Would be interesting to see 64bit userspace performance vs 32bit. Not sure why AsusWRT doesn't use 64bit in userspace. Wonder if it's to simplify userspace across all devices.

**Comment by lancethepants** [comment]
I've now got Tomatoware functioning on arm64.  Still fixing bugs as I discover them, but it seems to be pretty functional.  I made the switch to musl libc for arm64 because uclibc-ng isn't fully functioning on this architecture. Available right now in the nightly builds, let me know how it works for you.
https://files.lancethepants.com/Tomatoware/Nightly/08-17-2022/

**Comment by lancethepants** [comment]
Also got rust running as well.  Just `apt update` and `apt install rust`

**Comment by houzi-** [comment]
nice jod.I will try
